### PR TITLE
fix(azure-devops): strip HTML from description before storing

### PR DESCRIPTION
**Pull request by adityaharishch**
## Problem

Azure DevOps returns all rich-text fields (Description, Acceptance Criteria, Repro Steps) as raw HTML. We were storing it verbatim, so the UI was showing `<div>Bug test<br> </div>` and the LLM classifier was reading HTML markup as part of task content.

## Fix

Added `html_to_plaintext()` in `azure_devops.rs` — no new crate dependency:
- Strips all `<tag>` elements; `<br>`, `</p>`, `</div>`, `</li>` emit a space so words don't run together
- Decodes the 7 common HTML entities (`&amp;`, `&lt;`, `&gt;`, `&quot;`, `&#39;`, `&apos;`, `&nbsp;`)
- Collapses whitespace runs to a single space and trims

Applied at storage time (`force_refresh`) so both UI display and LLM classifier prompt receive clean text. Existing stored HTML will be fixed on the next sync.

## Test

Unit test `test_html_to_plaintext_basic` covers the exact failure case (`<div>Bug test<br> </div>` → `"Bug test"`), multi-div, entity decoding, list items, empty input.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by Akarsh-Hegde — COMMENTED** [review]
Automated review — 1 finding

File: src/intelligence/providers/azure_devops.rs [hunk 1]
```diff
@@ -78,6 +78,64 @@ struct StateDetail {
...
 }
 
+// ---------------------------------------------------------------------------
+// HTML helpers
+// ---------------------------------------------------------------------------
+
+/// Strip HTML tags and decode the most common entities so description fields
+/// stored in pm_tasks contain readable plain text rather than raw HTML markup.
+/// Azure DevOps returns HTML for Description, AcceptanceCriteria, ReproSteps, etc.
+fn html_to_plaintext(html: &str) -> String {
```
Comment (Akarsh-Hegde) [inline src/intelligence/providers/azure_devops.rs:88]: **Duplicate `html_to_plaintext` — PR #244 adds the same function to this file**

PR #244 (`feat/azure-devops-extra-fields`) also adds `fn html_to_plaintext` to this file, also targeting `main`. Whichever merges second will fail to compile with `error[E0201]: duplicate definitions with name 'html_to_plaintext'`.

**Recommended merge order:** merge this PR first (simpler fix, smaller diff), then rebase #244 on top and delete its copy of `html_to_plaintext` — it will already be present.
